### PR TITLE
feat(plugin-text-classification): BotonicPluginTextClassification implementation #BOT-342

### PR DESCRIPTION
**Pull request by elozano98**
## Description
Following the top-down approach, the first thing has been to create the plugin configuration and the main class: `BotonicPluginTextClassification`.
## Testing

The pull request...

- [ ] has unit tests
- [ ] has integration tests

### CODE REVIEW

File: packages/botonic-plugin-text-classification/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "test": "../../node_modules/.bin/jest",
+    "test_ci": "../../node_modules/.bin/jest --coverage --ci --reporters=default --reporters=jest-junit",
+    "prepare": "node ../../preinstall.js",
+    "lint": "npm run lint_core -- --fix",
+    "lint_ci": "npm run lint_core -- -c ../.eslintrc_slow.js",
+    "lint_core": "../../node_modules/.bin/eslint_d --cache --quiet 'src/**/*.ts*'"
+  },
+  "dependencies": {
+    "@babel/runtime": "^7.5.5",
+    "@botonic/core": "~0.18.0",
```
Comment (dpinol) [inline packages/botonic-plugin-text-classification/package.json:19]: try moving @botonic/core to devDependencies, as I think we only use types

File: packages/botonic-plugin-text-classification/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import type { PluginOptions } from './types'
+
+export default class BotonicPluginTextClassification implements Plugin {
+  constructor(readonly options: PluginOptions) {}
+
+  async pre(request: PluginPreRequest): Promise<void> {
+    try {
+      if (request.input.type == INPUT.TEXT && !request.input.payload) {
+        //   Pending to be implemented.
```
Comment (dpinol) [inline packages/botonic-plugin-text-classification/src/index.ts:12]: //TODO

File: packages/botonic-plugin-text-classification/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      }
+    } catch (e) {
+      console.log('Cannot classify the input text')
```
Comment (dpinol) [inline packages/botonic-plugin-text-classification/src/index.ts:15]: console.error('Cannot classify the input text', e)
NEVER hide an exception details. It's also nice to log the input (request.input.data?)

**Review by dpinol — CHANGES_REQUESTED** [review]

File: packages/botonic-plugin-text-classification/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export default class BotonicPluginTextClassification implements Plugin {
+  private modelsSelector: Promise<ModelSelector>
+
+  constructor(readonly options: PluginOptions) {}
+
+  async pre(request: PluginPreRequest): Promise<void> {
+    try {
+      if (request.input.type == INPUT.TEXT && !request.input.payload) {
+        if (request.input.type == INPUT.TEXT && !request.input.payload) {
```
Comment (dpinol) [inline packages/botonic-plugin-text-classification/src/index.ts:16]: line duplicated

File: packages/botonic-plugin-text-classification/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+          const inputText = request.input.data
+          const detectedLocale = detectLocale(inputText, this.options.locales)
+          const ner = (await this.modelsSelector).select(detectedLocale)
```
Comment (dpinol) [inline packages/botonic-plugin-text-classification/src/index.ts:19]: modelsSelector not set.

**Review by dpinol — CHANGES_REQUESTED** [review]

**Review by dpinol — APPROVED** [review]

**Review by vanbasten17 — APPROVED** [review]

**Pull request merged**